spyder/utils/qthelpers.py

```diff
@@ -502,7 +502,42 @@ def show_std_icons():
     dialog.show()
     sys.exit(app.exec_())
 
-    
+
+def create_control_layout(controls, main_widget):

```

Comment:
There would be a lot to say and discuss on the general UI design, but that's more in the scope of #4509.

I'm only trying to do a first small step here by:

1) extracting the currently common pattern of a single line of controls to a dedicated method. This is currently used sometimes in the widgets, sometimes in the plugin. I've intentionally not touched this to keep it simple.
2) Trying to find a way to to align this pattern with the tabbed editor.

I'll rewrite the patch to the simpler `create_control_layout(tools_layout, main_widget)` (better names are welcome, if you have one). Additionally, I'll limit the spacing modification to windows for the time being until I find a solution that's also working more generally.

That's not the solution to it alls, but one step better in both aspects. We can see further then.